python_modules/dagster/dagster/_core/storage/event_log/base.py

```diff
@@ -411,3 +411,48 @@ def alembic_version(self) -> Optional[AlembicVersion]:
     def is_run_sharded(self) -> bool:
         """Indicates that the EventLogStoarge is sharded."""
         return False
+
+    @property
+    def supports_global_concurrency_limits(self) -> bool:
+        """Indicates that the EventLogStorage supports global concurrency limits."""
+        return False
+
+    @abstractmethod
+    def allocate_concurrency_slots(self, concurrency_key: str, num_int: int) -> None:
+        """Allocate concurrency slots for the given concurrency key."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get_concurrency_limited_keys(self) -> Set[str]:
+        """Get the set of concurrency limited keys."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get_concurrency_info(self, concurrency_key: str, include_deleted: bool = False):
+        """Get concurrency info for key."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def claim_concurrency_slot(
+        self, concurrency_key: str, run_id: str, step_key: str, priority: Optional[int] = None
+    ):
+        """Claim concurrency slots for step."""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def free_concurrency_slots(self, run_id: str, step_key: Optional[str] = None):
```

Comment:
might be better to have one method for freeing a slot and one for clearing a whole run, just to reduce chance of accidental missing `step_key` clearing all bug

Reply:
+1